_posts/2020-11-09-pcr-primer-calculator-using-r.md

```diff
@@ -0,0 +1,298 @@
+![output file results]({{ site.url }}/assets/pcr-primer-calculator-using-r-output-file-results.jpg)
+
+Each output file will contain the entire PCR primer sequence (leading strand, restriction site and primer sequence), a count of each nucleotide of the *annealing* portion of the primer sequence, and the melting temperature of the primer sequence. Furthermore, we need to add in a minimum and maximum melting temperature section to the output file. I will explain why we need a minimum and maximum once we begin coding the nucleotide count code.
+
+-------------------------------------------------------------------------------
+## Creating our script in R
+#### Section 1: Adding libraries to the R script  
+
+Libraries were created to make coding easier for everyone. A library inside code is synonymous to a literal library. They are full of *books*. Each book you request from the library helps you accomplish a task. Imagine yourself as a computer, your human overlord demands that you take the integral of an infinite series that is both continuous and decreasing in a positive manor and needs to know if the function converges or diverges. What are you going to do? You will go to the library and request a Calculus textbook by Ron Larson. Once you have the book, you can tell your master the area under the curve of the infinite series is finite and will converge.
+
+For our script we will be using two libraries, readr and stringr.
+```r
+# load in the packages to help simplify our code
+library("readr")
+library("stringr")
+```
+
+#### Section 2: Import the PCR primer data
+
+**Overview**  
+2. **Import the PCR primer data**  
+  - Use readr to access the primer data from a csv files  
+  - Defensive coding: check the file for primer data   
+    - If no primers are found in csv file, abort the script  
+
+The readr library handles all the necessary coding for *streaming* in the input file into our script. now we can interacting with our primer data!
+```r
+input_file <- "path/to/pcr_primer_input_file.csv"
+primer_information <- readr::read_csv(input_file)
+# the primer data has been stored in variable primer_information
+# we will use this variable to access various parts of the data.
+```
+
+First we need to add a defensive coding block into our script to prevent empty input files from being processed. If we do not check for empty input files, then we might come across coding issues later on in the script. How do we prevent this? Lets count the number of rows in primer_information that have sequencing data.
+```r
+num_of_rows_in_data <- nrow(primer_information[, 1])
+# we will save this number to a variable becuase we will need it again within
+# our For Loop statment
+
+# defensive coding block
+if (num_of_rows_in_data <= 1) {
+  stop("ERROR: No sequence data found in the csv file")
+}
+```
+
+#### Section 3: Loop: Calculate the melting temperature of each primers
```

Comment:
Two colons like this bothers me (Section 3: Loop:). Try "Section 3: Use a loop to calculate the melting temperature of each primers." or something like that.